ee/clickhouse/queries/funnels/test/test_funnel.py

```diff
@@ -1400,3 +1402,64 @@ def test_funnel_with_elements_chain(self):
             self.assertCountEqual(
                 self._get_people_at_step(filter, 2), [person1.uuid],
             )
+
+    @snapshot_clickhouse_queries
+    def test_funnel_aggregation_with_groups(self):
+        GroupTypeMapping.objects.create(team=self.team, group_type="organization", group_type_index=0)
+        GroupTypeMapping.objects.create(team=self.team, group_type="company", group_type_index=1)
+
+        create_group(team_id=self.team.pk, group_type_index=0, group_key="org:5", properties={"industry": "finance"})
+        create_group(team_id=self.team.pk, group_type_index=0, group_key="org:6", properties={"industry": "technology"})
+
+        create_group(team_id=self.team.pk, group_type_index=1, group_key="company:1", properties={})
+        create_group(team_id=self.team.pk, group_type_index=1, group_key="company:2", properties={})
+
+        filters = {
+            "events": [
+                {"id": "user signed up", "type": "events", "order": 0},
+                {"id": "paid", "type": "events", "order": 1},
+            ],
+            "insight": INSIGHT_FUNNELS,
+            "aggregation_group_type_index": 0,
+            "date_from": "2020-01-01",
+            "date_to": "2020-01-14",
+        }
+
+        _create_person(distinct_ids=["user_1"], team_id=self.team.pk)
+        _create_event(
+            team=self.team,
+            event="user signed up",
+            distinct_id="user_1",
+            timestamp="2020-01-02T14:00:00Z",
+            properties={"$group_0": "org:5"},
+        )
+
+        # different person, same group, so should count as step two in funnel
+        _create_person(distinct_ids=["user_2"], team_id=self.team.pk)
+        _create_event(
+            team=self.team,
+            event="paid",
+            distinct_id="user_2",
+            timestamp="2020-01-03T14:00:00Z",
+            properties={"$group_0": "org:5"},
+        )
+
+        # same person, different group, so should count as different step 1 in funnel
+        _create_event(
+            team=self.team,
+            event="user signed up",
+            distinct_id="user_1",
+            timestamp="2020-01-10T14:00:00Z",
+            properties={"$group_0": "org:6"},
+        )
+
+        result = ClickhouseFunnel(Filter(data=filters), self.team).run()
```

Comment:
Note: Always pass `team` arg to filter object - otherwise simplification does not occur :)